### PR TITLE
Fix slices handling in LazyList

### PR DESCRIPTION
**Pull request by Bobronium**
### Motivation and context
Following [v2.16.2](https://github.com/cvat-ai/cvat/releases/tag/v2.16.2) release, we noticed an issue with annotations export with CVAT format and masks.

Upon investigation, we found that the problem stemmed from changes introduced in this PR: https://github.com/cvat-ai/cvat/pull/8229.

The issue was caused by the improper handling of slices with negative positions in LazyList, particularly affecting the computation of resulting data. This behaviour was triggered by the following line:
https://github.com/cvat-ai/cvat/blob/cfce7be5a9422da2c367296f56d70181185f503e/cvat/apps/dataset_manager/formats/cvat.py#L797

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
	- Enhanced handling of negative indices in list slicing, improving robustness and performance.
	- Added new test methods to validate slicing operations, ensuring correct behavior for various scenarios.

- **Bug Fixes**
	- Adjusted existing tests to account for empty values in the list, improving accuracy of expected outputs.

### CODE REVIEW

**Comment by zhiltsov-max** [comment]
Could you also update test assets? I can see there are changes in array representations - the brackets are removed.

**Review by zhiltsov-max — COMMENTED** [review]
- I suggest that the PR is renamed, as the problem is not directly related to the CVAT format
- Please update changelog

File: cvat/apps/engine/tests/test_lazy_list.py [hunk 1]
```diff
@@ -170,14 +165,31 @@ def test_length_on_iteration(self):
...
+        self.assertEqual(self.lazy_list[:-2], [1])
+        self.assertEqual(self.lazy_list[:-1], [1, 2])
+        self.assertEqual(self.lazy_list[-2::-1], [2, 1])
+
+    def test_slice_with_stride(self):
+        self.assertEqual(self.lazy_list[::2], [1, 3])
+        self.assertEqual(self.lazy_list[::-2], [3, 1])
+        self.assertEqual(self.lazy_list[1::-1], [2, 1])
+        self.assertEqual(self.lazy_list[1:3:2], [2])
+
```
Comment (zhiltsov-max) [inline cvat/apps/engine/tests/test_lazy_list.py:192]: It looks like it's better to split these test cases further or merge them together. I'd also added a more long list and an empty one.

Comment (Bobronium) [reply cvat/apps/engine/tests/test_lazy_list.py]: Good point. I've unified this in one test case, but splitted each one in a separate subtest.

And made list longer for all test cases.

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**